### DISCUSSION TITLE
Do we have to add a manifest every time a new version of the program is released?

### DISCUSSION THREAD

**Discussion by kneelesh48** [microsoft/winget-pkgs#16014] [Q&A] [answered]
Do we have to add a manifest every time a new version of the program is released?

If that's the case, there's potential that a lot of packages found on winget will be outdated.
People are actively adding programs to winget because it is new and shiny.
But soon the motivation will wear off and maintainers will be reluctant to update.
People will prefer not to use winget because it's likely to download an outdated version of the program.

I think the process of adding manifests of new version can be automated by some kind of crawler bot.

What are your thoughts?

**Comment by denelon** [comment, marked as answer]
@kneelesh48 we do have automation to check for expired URLs and invalid SHA256 hashes. Some scenarios have already been automated to automatically detect new versions, and submit PRs for the updated manifests.

There are still other cases that haven't been automated. We have some work items in the backlog for those like when the source is a GitHub project, and we could possibly use a webhook or something similar. There are also packages with RSS feeds identifying latest versions. The level of automation support we can provide is limited due to some more complex publishing/upgrading strategies, but we will tackle what we can. Be sure to add your 👍 to those Issues.
